jerry-core/lit/lit-char-helpers.c

```diff
@@ -894,3 +897,51 @@ lit_char_to_upper_case (lit_code_point_t cp, /**< code point */
   return cp;
 #endif /* ENABLED (JERRY_UNICODE_CASE_CONVERSION) */
 } /* lit_char_to_upper_case */
+
+#if ENABLED (JERRY_ESNEXT)
+/*
+ * Look up whether the character should be folded to the lowercase variant.
+ *
+ * @return true, if character should be lowercased
+ *         false, otherwise
+ */
+bool
+lit_char_fold_to_lower (lit_code_point_t cp) /**< code point */
+{
+#if ENABLED (JERRY_UNICODE_CASE_CONVERSION)
+  return (cp > LIT_UTF16_CODE_UNIT_MAX
+          || (!lit_search_char_in_interval_array ((ecma_char_t) cp,
+                                                  lit_unicode_folding_skip_to_lower_interval_starts,
+                                                  lit_unicode_folding_skip_to_lower_interval_lengths,
+                                                  NUM_OF_ELEMENTS (lit_unicode_folding_skip_to_lower_interval_starts))
+              && !lit_search_char_in_array ((ecma_char_t) cp,
+                                            lit_unicode_folding_skip_to_lower_chars,
+                                            NUM_OF_ELEMENTS (lit_unicode_folding_skip_to_lower_chars))));
+#else /* !ENABLED (JERRY_UNICODE_CASE_CONVERSION) */
+  return true;
```

Comment:
The return value of `lit_char_fold_to_lower` and `lit_char_fold_to_upper` is different when  `JERRY_UNICODE_CASE_CONVERSION` is not enabled. It looks odd to me. Is it intentional?

Reply:
Yes, 1 byte code points always fold to the lowercase variant when unicode mode is enabled, that's why fold to upper is false.